scapy/packet.py

```diff
@@ -771,7 +771,8 @@ def guess_payload_class(self, payload):
         """
         for t in self.aliastypes:
             for fval, cls in t.payload_guess:
-                if all(hasattr(self, k) and v == self.getfieldval(k)
+                if all(((hasattr(self, k) and v == self.getfieldval(k)) or
+                        (k == 'payload_dispatcher' and (hasattr(v, '__call__') or callable(v)) and v(self, payload)))
```

Comment:
I assume this checks a condition before doing the binding, instead of using that, you would rather add a custom `guess_payload_class` to the packet. You can return Raw or Padding, or even else in the case its not wanted

Reply:
If this works as I expect after your short explanation this is
a. wesome
b. etter way to solve my problem
c. leary something that should go to the scapy docs

Reply:
addressed in  43c00aa (revert change @ guess payload) and 867ae6c (use dispatch_hook)

Reply:
also created a gist that outlines the usage of dispatch_hook (https://gist.github.com/speakinghedge/9b70073695ec1ffd9a517eb8e344e388). Could you please have a look at it? If I got everything right regarding the hook I would like to extend scapys documentation using this example...

Reply:
Yep, your usage is correct ! 

It’s not compulsory for the layers to be identical except one byte, nor for them to have the same payload, but that’s mostly correct.

The big difference between dispatch_hook and guess_payload is that it allows to have a generic class, that you call (as a user) to dissect your packet, that will select the appropriate packet that actually matches the data

Reply:
Thanks for the review